lib/elixir/lib/string.ex

```diff
@@ -1756,30 +1756,48 @@ defmodule String do
 
   """
   @spec chunk(t, :valid | :printable) :: [t]
+  def chunk(string, trait) when is_binary(string) and trait in [:valid, :printable] do
+    chunk_by(string, make_chunk_pred(trait))
+  end
+
+  @doc ~S"""
+  Splits the string on every character for which `fun` returns a new value.
 
-  def chunk(string, trait)
+  Returns a list of binaries in which each character produces the same output
+  when passed to `fun`.
 
-  def chunk("", _), do: []
+  ## Examples
 
-  def chunk(string, trait) when is_binary(string) and trait in [:valid, :printable] do
+      iex> String.chunk_by(" foo   bar ", fn cp -> cp =~ "\s" end)
+      [" ", "foo", "   ", "bar", " "]
+
+      iex> String.chunk_by("Mississippi", &Function.identity/1)
+      ["M", "i", "ss", "i", "ss", "i", "pp", "i"]
+
+      iex> String.chunk_by("Ńaïve", fn cp -> cp < "~" end)
```

Comment:
I think we should remove this example. Comparison < is most likely not going to behave how people expect, especially for Unicode.